cmd/queue/main.go

```diff
@@ -61,6 +60,10 @@ const (
 	// in the mesh.
 	quitSleepDuration = 20 * time.Second
 
+	// Set equal to the queue-proxy's ExecProbe timeout to take
+	// advantage of the full window
+	queueProbeTimeout = 10 * time.Second
```

Comment:
Please use this in the actual `probeUserContainer` as well to make sure all of these values are properly aligned.

Reply:
I wasn't sure whether it was suitable to use it there since `probeUserContainer` was also used in another handler and I didn't feel confident totally tying the concepts together.

How is the `K-Network-Probe` header used by the system with respect to the queue-proxy?

If we don't reuse the same probe in the `/health` check and the `K-Network-Probe`, we can leave the network probe's functionality as-is and then potentially remove the retry/timeout logic from the `/health` "server-side" TCP probe. Having all of the `readinessProbe` retry interval/timeout logic live in a single place on the `-probe` HTTP "client" side of things would make it easier to read/reason about. There would be a little bit of duplicated code and each retried TCP probe would then also have the additional overhead of an HTTP roundtrip. Thoughts?

If not, should I just make a duplicate `probeUserContainer` and use the constant there?

And I'd definitely appreciate someone else booting it up and taking it for a spin. 😁

Reply:
Not sure I'm following. The inner TCP loop currently has exactly this 10s timeout as well and those timeouts are meant to align, I think?

Reply:
Totally agree that they should align and I guess that's a great reason for me to stop complaining and make the change. My reluctance came from the fact that `probeUserContainer` is also used by `handler` for what appears to be another purpose, though I guess that can be worried about when it becomes an issue. I will make the swap now. 👍

Reply:
@markusthoemmes @greghaynes
Is the above-mentioned tracing tool available somewhere that I can run myself?

Reply:
@joshrider https://github.com/knative/serving/pull/3491 will give you the most level of detail if you're able to rebase on it. Otherwise using just master if you enable tracing in config/config-tracing.yaml you should see a more consistent / smaller probe span in zipkin